### PR TITLE
MULE-19100 : Provide a reusable mechanism for using object fields as acting parameters

### PR DESCRIPTION
**Pull request by ndinu**

### CODE REVIEW

**Comment by ndinu** [comment]
--validate

**Review by marianogonzalez — APPROVED** [review]

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/InjectableParameterResolver.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    parameterValue = resolvedParameterValues.get(parameterName);
+    InjectableParameterInfo injectableParameterInfo = injectableParametersMap.get(parameterName);
+
+    if (parameterValue != null) {
+      try {
+        parameterValue = expressionManager
+            .evaluate("#[payload]",
+                      DataType.fromType(getClassFromType(injectableParameterInfo.getType())),
+                      BindingContext.builder()
+                          .addBinding("payload", new TypedValue(parameterValue, DataType.fromObject(parameterValue))).build())
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/InjectableParameterResolver.java:78]: what about the mediatType? Doesnt the `fromObject` method return  `application/java` for a String containing a Json. Are there any tests using this with json payloads?

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/InjectableParameterResolver.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+            .getValue();
+      } catch (ClassNotFoundException e) {
+        LOGGER.debug("Transformation of injectable parameter '{}' failed, the same value of the resolution will be used.",
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/InjectableParameterResolver.java:81]: why is a class cast something that you can just ignore and continue as is? Shouldn't this actually raise an error to be dealt with?

Comment (ndinu) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/InjectableParameterResolver.java:81]: It is a class not found exception. If the current class loader does not have the class being looked for, we will do a best effort and try to inject the value as is to the value provider.

**Pull request merged**